pkg/machines/components/vmDisksTab.jsx

```diff
@@ -106,13 +107,27 @@ const VmDisksTab = ({ idPrefix, vm, disks, actions, renderCapacity, dispatch, pr
                     columns.push(disk.diskExtras);
 
                     if (provider === 'LibvirtDBus') {
-                        const removeDiskAction = RemoveDiskAction({
-                            dispatch,
-                            vm,
-                            target: disk.target,
-                            idPrefixRow,
-                            onAddErrorNotification,
-                        });
+                        const onDetachDisk = () => {
+                            return () => {
+                                dispatch(detachDisk({ connectionName:vm.connectionName, id:vm.id, name:vm.name, target: disk.target, live: vm.state == 'running' }))
```

Comment:
Do you need the dispatch wrapper here?

Reply:
It seems so, otherwise it would result in TypeError: Object(...)(...).catch is not a function .